src/main/java/com/fasterxml/jackson/databind/deser/std/FromStringDeserializer.java

```diff
@@ -330,5 +335,31 @@ protected int _firstHyphenOrUnderscore(String str)
             }
             return -1;
         }
+
+        /** see {@link Locale#toString()}, but ignore extends */
+        private Locale deJavaLocale(String value) {
+            if (value.length() == 0) {
+                return new Locale("");
+            }
+
+            int extStart = value.indexOf("_#");
+            if (extStart != -1) value = value.substring(0, extStart);
+
+            int pos1 = value.indexOf('_');
+            if (pos1 == -1) { // It is not java;
+                return null;
```

Comment:
No, that's just plain wrong. You can not arbitrarily return `null` for something unrecognized: that should return in an exception.